src/graphics/assets_manager.c

```diff
@@ -0,0 +1,122 @@
+#include "assets_manager.h"
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define BUF_SIZE 200
+
+/**
+ * Unload all existing textures
+ */
+void _unload_textures(AssetManager *am);
+
+/**
+ * Duplicate a string onto the heap.
+ */
+char *_duplicate_string(char *string);
+
+AssetManager *initAssetManager() {
+    AssetManager *am = malloc(sizeof(AssetManager));
+    am->textures_loaded = false;
+    am->theme = NULL;
+    return am;
+}
+
+char *_duplicate_string(char *string) {
+    if (string == NULL)
+        return NULL;
+
+    int length = strlen(string);
+    char *new = malloc(sizeof(char) * (length + 1));
+    if (new == NULL)
+        return NULL;
+
+    for (int i = 0; i < length; i++) {
+        new[i] = string[i];
+    }
+
+    new[length] = '\0';
+    return new;
+}
+
+void loadAllAssets(AssetManager *assets, char *theme) {
+    if (assets->textures_loaded) {
+        // if the current theme is loaded, do nothing
+        if (strcmp(theme, assets->theme) == 0) {
+            return;
+        }
+
+        _unload_textures(assets);
+    }
+
+    char *theme_cpy = _duplicate_string(theme);
+    assets->theme = theme_cpy;
+
+    char buffer[BUF_SIZE];
+
+    snprintf(buffer, BUF_SIZE, "assets/themes/%s/%s.png", theme, "tile-1");
```

Comment:
Perhaps a method/function can be made to reduce the repetition here

Reply:
I actually wanted to put this into a loop but the fact that each member of the struct is named separately, I have to explicitly write it out (`assets->tile-1`). This would also be the same problem with extracting this to a function.